frontend/lib/wallets/MetaMask.ts

```diff
@@ -31,6 +39,12 @@ export default class MetaMask extends AbstractWallet {
         if (!window.ethereum) {
             return false;
         }
+        if (window.ethereum.providers?.length) {
+            const metaMask = window.ethereum.providers.find((p: any) => p.isMetaMask);
```

Comment:
nit: i'm a strong proponent of complete names. Hence i suggest to name it `provider` instead of `p`